src/conclusion_stage_image.py

```diff
@@ -0,0 +1,68 @@
+"""
+Copyright 2019-2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License"). You
+may not use this file except in compliance with the License. A copy of
+the License is located at
+
+    http://aws.amazon.com/apache2.0/
+
+or in the "license" file accompanying this file. This file is
+distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+ANY KIND, either express or implied. See the License for the specific
+language governing permissions and limitations under the License.
+"""
+
+from image import DockerImage
+from context import Context
+from utils import generate_safety_report_for_image
+
+import os
+
+class ConclusionStageImage(DockerImage):
+    """
+    Class designed to handle the ConclusionStageImages
+    """
```

Comment:
It makes sense that it is the "last stage" of the docker build, but this code seems like it can be reused for other kinds of multi-stage builds. Besides that, we don't know for sure if this will always be the last stage of a docker image build forever. We may add more layers in the future.

The issue here is that a small decision like that would result in the code either becoming confusing because "Conclusion" is no longer the last stage, or having to be revamped pretty heavily to change the names of classes and objects related to this. I would like to make this future proof by not making the assumption that the safety report layer will always be the "concluding" stage.

Reply:
Sure. That should be good. Please also make add this explanation to the docstring for the class, so that it is readily apparent what the class is meant to do.

Reply:
Done

Reply:
@shantanutrip Can you explain the reason behind `I agree that we should remove the assumption that this is the last stage of the docker build.` ? This docker layer is meant to be the final stage in the build process as per the design discussions. I agree that we can add more layers on top of the common layer but what is the usecase for that?

Ideally, we want the common components to run at the very last stage of the build. Eg, we want to run ECR scan/safety report/oss compliance licenses on the final layer of the docker build so that there are no new components in the docker added after this final layer.